test/integration/roles/test_git/tasks/main.yml

```diff
@@ -435,6 +437,60 @@
   assert:
     that: "repo_content.stat.exists"
 
+# See https://github.com/ansible/ansible-modules-core/pull/4243
+
+- name: clear checkout_dir
+  file: state=absent path={{ checkout_dir }}
+
+- name: clone repo
+  git: repo={{ repo_update_url_1 }} dest={{ checkout_dir }}
+
+- name: clone repo with same url to same destination
+  git: repo={{ repo_update_url_1 }} dest={{ checkout_dir }}
+  register: checkout_same_url
+
+- name: check repo not changed (when get url command supported in git version)
+  assert:
+    that:
+      - not checkout_same_url|changed
+  when: git_version.stdout | version_compare("{{git_version_supporting_get_url}}", '>=')
+
+- name: check repo changed (when get url command not supported in git version)
```

Comment:
I would not ensure that the bug is there. Only test for desired behavior. Otherwise someone fixing this behavior in the future will make these tests fail.